src/app/static/src/app/store/modelOptions/mutations.ts

```diff
@@ -18,5 +20,16 @@ export const mutations: MutationTree<ModelOptionsState> & MetadataMutations = {
 
     update(state: ModelOptionsState, payload: DynamicFormMeta) {
         state.optionsFormMeta = payload;
+    },
+
+    FormMetaUpdated(state: ModelOptionsState, action: PayloadWithType<DynamicFormMeta>) {
```

Comment:
There's a possible workflow that looks like:
1. user uploads survey data
2. user proceeds to model options step and selects some options
3. user returns to upload step and uploads program data

In this case we want to append the new section returned from the model options endpoint (some sections are conditional on program data being present) but we don't want to overwrite the already selected options. Hence this mutation which appends or removes sections (once https://vimc.myjetbrains.com/youtrack/issue/mrc-657 is in users will also be able to remove uploaded files, so could go from program + survey -> survey only), but does not overwrite selected values.

See tests for this here: https://github.com/mrc-ide/hint/pull/139/files#diff-97445ba69387f1718e7c5f330133e49dR20

Reply:
And the control sections themselves shouldn't change structure in the course of mucking about with different files? They'll always have the same controls and options if they're present at all?

Reply:
This is a good question. At the moment yes, the only thing that changes is whether or not the ART and ANC sections are included or not. But I'll check with Rob if that could change. If so, we'd have to just accept that options will get overwritten if you change your input files.

Reply:
"Yeah internal structure of each section is fixed. Only expecting ART and ANC to be possibly missing.
Whether it is likely to change I'm not sure, there isn't a mechanism for that in the R side atm. Only able to exclude/include at the section level.
I'd have to check with Jeff though for sure"

Reply:
Following discussion today it seems that controls within sections *will* change. And we'll have to re-request model options every time the user lands on the model options page. Which means we probably don't want to request them when files are uploaded, as I've done on this branch. So if you hold off re-reviewing this, I'll make those changes on this branch.